packages/flutter_tools/gradle/build.gradle.kts

```diff
@@ -3,7 +3,21 @@
 // found in the LICENSE file.
 
 plugins {
+    `java-gradle-plugin`
     `groovy-gradle-plugin`
+    kotlin("jvm") version "1.8.10"
+    `kotlin-dsl`
+}
+
+buildscript {
+    repositories {
+        google()
+        mavenCentral()
+    }
+
+    dependencies {
+        classpath("com.android.tools.build:gradle:7.3.0")
```

Comment:
If it does not cause issues consider making this 7.4.2 which I think is the highest 7.* version published.

Reply:
+1

Reply:
I tried in [`0e7e128`](https://github.com/flutter/flutter/pull/127897/commits/0e7e1282bf5b7032dddd0cf1c7718264f535fb99), but CI failed because of those changes.

Let's bump this version in some another PR.